br/pkg/streamhelper/advancer.go

```diff
@@ -424,7 +424,15 @@ func (c *CheckpointAdvancer) onTaskEvent(ctx context.Context, e TaskEvent) error
 		c.task = e.Info
 		c.taskRange = spans.Collapse(len(e.Ranges), func(i int) kv.KeyRange { return e.Ranges[i] })
 		c.setCheckpoints(spans.Sorted(spans.NewFullWith(e.Ranges, 0)))
-		c.lastCheckpoint = newCheckpointWithTS(e.Info.StartTs)
+		globalCheckpointTs, ok := c.env.GetGlobalCheckpointForTask(ctx, c.task.Name)
+		if ok != nil {
+			log.Warn("Advancer failed to get global checkpoint")
+			globalCheckpointTs = 0
```

Comment:
Once fail to get global checkpoint here. the task may very easy to stop. can we retry for this error here? with `utils.Retry`

Reply:
I guess we can directly return error here?

Reply:
> Once fail to get global checkpoint here. the task may very easy to stop. can we retry for this error here? with `utils.Retry`

added